src/main/scala/chisel3/stage/ChiselOptions.scala

```diff
@@ -7,20 +7,22 @@ import chisel3.internal.WarningFilter
 import java.io.File
 
 class ChiselOptions private[stage] (
-  val printFullStackTrace: Boolean = false,
-  val throwOnFirstError:   Boolean = false,
-  val outputFile:          Option[String] = None,
-  val chiselCircuit:       Option[Circuit] = None,
-  val sourceRoots:         Vector[File] = Vector.empty,
-  val warningFilters:      Vector[WarningFilter] = Vector.empty) {
+  val printFullStackTrace:   Boolean = false,
+  val throwOnFirstError:     Boolean = false,
+  val outputFile:            Option[String] = None,
+  val chiselCircuit:         Option[Circuit] = None,
+  val sourceRoots:           Vector[File] = Vector.empty,
+  val warningFilters:        Vector[WarningFilter] = Vector.empty,
+  val legacyShiftRightWidth: Boolean = false) {
```

Comment:
Nit: the naming of this may be better as `useLegacyShiftRightWidth`, i.e., something which is more obviously true/false.